### PR TITLE
AKS: implement pod identity addons management sub commands

### PR DESCRIPTION
**Pull request by bcho**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

**Comment by bcho** [comment]
@fengzhou-msft @arrownj could you review this PR? It's ready to review & merge now!

File: src/aks-preview/azext_aks_preview/_help.py [hunk 1]
```diff
@@ -281,6 +281,9 @@
         - name: --linux-os-config
           type: string
           short-summary: OS configurations for Linux agent nodes.
+        - name: --enable-pod-identity
+          type: bool
+          short-summary: Enable pod identity addon.
```
Comment (levimm) [inline src/aks-preview/azext_aks_preview/_help.py:286]: also need a PREVIEW here?

Comment (bcho) [reply src/aks-preview/azext_aks_preview/_help.py]: updated, PTAL

File: src/aks-preview/azext_aks_preview/_help.py [hunk 2]
```diff
@@ -528,7 +541,7 @@
 
 helps['aks nodepool'] = """
     type: group
-    short-summary: Commands to manage node pools in Kubernetes kubernetes cluster.
+    short-summary: Commands to manage node pools in managed Kubernetes cluster.
```
Comment (zqingqing1) [inline src/aks-preview/azext_aks_preview/_help.py:544]: thanks for correct this. 👍

File: src/aks-preview/azext_aks_preview/_params.py [hunk 3]
```diff
@@ -202,6 +206,52 @@ def load_arguments(self, _):
...
                    default=os.path.join(os.path.expanduser('~'), '.kube', 'config'))
 
+    with self.argument_context('aks pod-identity') as c:
+        c.argument('cluster_name', type=str, help='The cluster name.')
+
+    with self.argument_context('aks pod-identity add') as c:
+        c.argument('identity_name', type=str, options_list=['--name', '-n'], default=None, required=False,
+                   help='The pod identity name. Generate if not specified.',
+                   validator=validate_pod_identity_resource_name('identity_name', required=False))
+        c.argument('identity_namespace', type=str, options_list=['--namespace'], help='The pod identity namespace.')
```
Comment (zqingqing1) [inline src/aks-preview/azext_aks_preview/_params.py:216]: Is the namespace "Required"?

**Review by zqingqing1 — CHANGES_REQUESTED** [review]

File: src/aks-preview/azext_aks_preview/_params.py (hunk 3 above)
Comment (bcho) [reply src/aks-preview/azext_aks_preview/_params.py:216]: yes, it's required

Comment (bcho) [reply src/aks-preview/azext_aks_preview/_params.py:216]: the name is optional

File: src/aks-preview/azext_aks_preview/commands.py [hunk 4]
```diff
@@ -70,3 +72,15 @@ def load_command_table(self, _):
...
         g.custom_command('delete', 'aks_agentpool_delete', supports_no_wait=True)
         g.custom_command('get-upgrades', 'aks_agentpool_get_upgrade_profile')
+
+    # AKS pod identity commands
+    with self.command_group('aks pod-identity', managed_clusters_sdk, client_factory=cf_managed_clusters) as g:
+        g.custom_command('add', 'aks_pod_identity_add')
+        g.custom_command('delete', 'aks_pod_identity_delete')
+        g.custom_command('list', 'aks_pod_identity_list',
+                         table_transformer=aks_pod_identities_table_format)
+        g.custom_command('add-exception', 'aks_pod_identity_exception_add')
```
Comment (arrownj) [inline src/aks-preview/azext_aks_preview/commands.py:82]: We suggest to add a command group for exception here. The commands should be like:
1. `exception add`
2. `exception delete`
3. `exception update`
4. `exception list`

And maybe an additional `exception show` is also needed ?

Comment (bcho) [reply src/aks-preview/azext_aks_preview/commands.py:82]: we don't have `exception show`. I could create a new sub group `pod-identity exception add/delete/update/list` cc @miwithro

Comment (bcho) [reply src/aks-preview/azext_aks_preview/commands.py]:
```
az aks pod-identity exception -h

Group
    az aks pod-identity exception : Commands to manage pod identity exceptions in managed Kubernetes
    cluster.

Commands:
    add    : Add a pod identity exception to a managed Kubernetes cluster.
    delete : Remove a pod identity exception from a managed Kubernetes cluster.
    list   : List pod identity exceptions in a managed Kubernetes cluster.
    update : Update a pod identity exception in a managed Kubernetes cluster
```

Updated, PTAL @arrownj

**Review by arrownj — APPROVED** [review]
LGTM

**Comment by bcho** [comment]
@zqingqing1 could you help review & approve this pr? We need your approval vote. Thanks!

**Review by zqingqing1 — APPROVED** [review]
lgtm

**Pull request merged**